### DISCUSSION TITLE
Dedupe Eslint config based on extended configs

### DISCUSSION THREAD

**Discussion by Lioness100** [eslint/eslint#17552] [Q&A]
I'd like to programmatically remove redundant rules in my eslint config if the rule has been configured the same in a config I'm extending. Is there any tool that already does this? If not, could you give me a pointer for how I should create a tool myself? I'm not familiar with the internals of eslint. Thanks!

**Comment by Lioness100** [comment]
Ok, I've figured out a little more of what I'm doing. I used `FlatCompat` from `@eslint/eslintrc` to load and transform my config into a flat config. Then, I input it into a `FlatConfigArray` instance from `eslint` (which I had to manually export - not sure how I'd do this in practice without recreating the class with `ConfigArray` from `@humanwhocodes/config-array`). Then, I normalized it, and called `getConfig()` for a random JSON file. I manually edited https://github.com/eslint/eslint/blob/55c1685aab0fa75074ac1033abfa6e34cb7baaf6/lib/config/flat-config-schema.js so that it would log if either schemas in `rulesSchema.merge()` shared an identical rule. Again, not sure how I'd do this in practice.

Here are my questions:

* Is there any public API to find the eslint config file?
* Is there any public API that will return a flat config array from a config file, regardless of whether it uses the flat or legacy format?
* How can I access and modify the private objects like `rulesSchema` and `FlatConfigArray`?
* Since `getConfig()` requires a file which in then matches to the `files` or `ignore` fields, I'm not sure what I'd input in there. Should I remove the `files` and `ignore` fields from the config? But then it might merge with rules it normally didn't before...
* This also triggers if two extended configs have conflicts with each other, which I don't care about. Maybe the best strategy would be to merge the extended configs separately with the usual settings and then merge *that* with the user's settings while reporting duplicates. Easier said than done...

**Comment by Rec0iL99** [comment]
Hi @Lioness100, you might want to take a look at https://github.com/eslint/eslint/issues/5116. 

Our [Nodejs API](https://eslint.org/docs/head/integrate/nodejs-api) is a good starting point.

https://eslint.org/docs/latest/integrate/nodejs-api#-eslintcalculateconfigforfilefilepath
